kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -406,14 +196,22 @@ public open class ApplicationCommandRegistry : KoinComponent {
 
                 null
             }
-        }.filterNotNull()
-
-    // endregion
+        }
 
-    // region: Typed registration functions
+    /**
+     * Creates a KordEx [ApplicationCommand] as discord command and returns the created command's id as [Snowflake].
+     */
+    public open suspend fun createDiscordCommand(command: ApplicationCommand<*>): Snowflake? = when (command) {
+        is SlashCommand<*, *> -> createDiscordSlashCommand(command)
+        is UserCommand<*> -> createDiscordUserCommand(command)
+        is MessageCommand<*> -> createDiscordMessageCommand(command)
+        else -> throw IllegalArgumentException("Unknown ApplicationCommand type")
```

Comment:
Again, don't make this exhaustive. We want a compiler error here.

Reply:
It needs to be exhaustive, it won't compile without it.